crates/runc/src/monitor.rs

```diff
@@ -72,9 +81,87 @@ impl DefaultMonitor {
     }
 }
 
+/// Process exit status returned by [ProcessMonitor::wait()].
 #[derive(Debug)]
 pub struct Exit {
     pub ts: OffsetDateTime,
     pub pid: u32,
     pub status: i32,
 }
+
+/// Execute a `Command` and collect exit status, output and error messages.
+///
+/// To collect output and error messages, pipes must be used for Command's stdout and stderr.
+/// This method will create pipes and overwrite stdout/stderr of `cmd`.
+///
+/// Note: invalid UTF-8 characters in output and error messages will be replaced with the `�` char.
+pub async fn execute<T: ProcessMonitor + Send + Sync>(
+    monitor: &T,
+    mut cmd: Command,
+) -> Result<(Exit, ExitStatus, String, String), Error> {
```

Comment:
May be introduce a struct? Too many return params.

Reply:
Fixed